cli/cmd/install-service-mirror.go

```diff
@@ -92,6 +92,10 @@ func (options *installServiceMirrorOptions) validate() error {
 		return errors.New("you need to specify a namespace")
 	}
 
+	if options.namespace == controlPlaneNamespace {
+		return errors.New("you need to install the service mirror component in a namespace different than the Linkerd one")
+	}
```

Comment:
Note that for this check to be effective it requires that the user passes the `--linkerd-namespace` flag if it's not the default "linkerd". That, or having this command fetch `linkerd-config` CM.

Reply:
So do you suggest I actually check for the presence of the conf map instead ?

Reply:
Yeah, IIUC we're coupling the service mirror controller with a linkerd control plane after all, so we could do just like with `linkerd cluster setup-remote` and check for the CM

Reply:
Yes, makes sense! That is what I will do